src/type_.rs

```diff
@@ -120,8 +120,13 @@ impl<'gcc, 'tcx> BaseTypeMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
         self.isize_type
     }
 
+    #[cfg(feature = "master")]
     fn type_f16(&self) -> Type<'gcc> {
-        unimplemented!("f16_f128")
+        self.context.new_c_type(CType::Float16)
+    }
```

Comment:
I'm surprised the CI didn't complain about formatting, but it's missing an empty line between the two methods.